Float.TinCan.ActivityLibrary/ActivityLaunchCoordinator.cs

```diff
@@ -318,19 +332,22 @@ protected virtual void HandleAssessmentFailed(object sender, EventArgs args)
         /// </summary>
         /// <param name="sender">The sending object.</param>
         /// <param name="args">Arguments related to the event.</param>
-        protected virtual void HandleDownloadCompleted(object sender, EventArgs args)
+        protected virtual async void HandleDownloadCompleted(object sender, EventArgs args)
         {
             if (downloadStatus != null && downloadStatus.State != DownloadStatus.DownloadState.Error)
             {
                 downloadStatus.DownloadsCompleted -= HandleDownloadCompleted;
                 downloadStatus.DownloadsCancelled -= HandleDownloadCancelled;
 
-                NavigationContext.DismissPage();
                 startLocation = Activity.MetaData.StartLocation;
+                isCreatingRunner = true;
+
+                await NavigationContext.DismissPageAsync();
```

Comment:
Before merging, since this will require a newer version of Float.Core, the csproj file needs to be updated to document the minimum version of Float.Core needed here.